sqlmesh/core/snapshot/definition.py

```diff
@@ -928,6 +931,9 @@ def categorize_as(self, category: SnapshotChangeCategory) -> None:
                     previous_version.data_version.temp_version
                     or previous_version.fingerprint.to_version()
                 )
+        elif self.is_model and self.model.forward_only and not self.previous_version:
+            # If this is a new model then use a deterministic version, independent of the fingerprint.
+            self.version = hash_data([self.name, *self.model.kind.data_hash_values])
```

Comment:
so the only way this would be problematic is if two people made new forward only models that aren't compatible

Reply:
Depends on what you mean by being incompatible?

Reply:
i make a branch model A with column x

you make a branch with model b with column y

this now crashes right?

Reply:
Yes, I've just thought of this too. It actually shouldn't, because the table migration should take care of this but it only kicks in for `FORWARD_ONLY` categories. However, both new model versions will be categorized as `BREAKING`. So I need to make sure that the migration is invoked, and we should be good even in this scenario.

Good catch.

Reply:
Going to add an integration test for this

Reply:
Added an integration test for this